pkg/sippyserver/server.go

```diff
@@ -265,6 +269,132 @@ func (s *Server) detailed(w http.ResponseWriter, req *http.Request) {
 
 }
 
+type Tests map[string][]int
+
+func unpackTests(tests []testgridv1.Test) Tests {
+	result := make(Tests)
+	for _, t := range tests {
+		var statuses []int
+		for _, testResult := range t.Statuses {
+			for i := 0; i < testResult.Count; i++ {
+				statuses = append(statuses, testResult.Value)
+			}
+		}
+		result[t.Name] = statuses
+	}
+	return result
+}
+
+func hasOpenShiftTests(tests Tests, i int) bool {
+	for test, results := range tests {
+		if strings.HasPrefix(test, "openshift-tests.") {
+			if results[i] != 0 {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func testStatus(tests Tests, name string, i int) int {
+	results, ok := tests[name]
+	if !ok {
+		return 0
+	}
+	return results[i]
+}
+
+func (s *Server) jobs(w http.ResponseWriter, req *http.Request) {
+	reportName := req.URL.Query().Get("release")
+	jobFilterString := req.URL.Query().Get("jobFilter")
+
+	var jobFilter *regexp.Regexp
+	if len(jobFilterString) > 0 {
+		var err error
+		jobFilter, err = regexp.Compile(jobFilterString)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("jobFilter: %s", err), http.StatusBadRequest)
+			return
+		}
+	}
+
+	dashboardCoordinates, found := s.reportNameToDashboardCoordinates(reportName)
+	if !found {
+		releasehtml.WriteLandingPage(w, s.reportNames())
+		return
+	}
+
+	type jsonTest struct {
+		Name    string   `json:"name"`
+		Results []string `json:"results"`
+	}
+	type jsonJob struct {
+		Name        string     `json:"name"`
+		Timestamps  []int      `json:"timestamps"`
+		Tests       []jsonTest `json:"tests"`
```

Comment:
this array doesn't align w/ the other arrays in this struct.

All the other arrays contain elements that represent "instance data for an instance of the job"

so it makes the api a bit awkward.

Reply:
This is a leftover from the first iterations, I was thinking about returning "Overall" and some selected tests.
Updated.